### PR TITLE
Dragpoints CustomEditor upgrades.

### PR DESCRIPTION
**Pull request by Vroonsh**
- Drag points selection & handling management
- Drag points multi selection using Ctrl key
- Rendering Drag points, curve & curve traveller
- Drag point lock management
- Popup menu on both drag points & curve traveller
- Switching drag  point booleans from popup menu (IsLocked, IsSlingshot, IsSmooth)
- Drag point Add/Remove management using popup menu (on DP or curve traveller)
- Fixed mesh rebuild graphic glitches.
- Fixed Triggers wrong drag points positions.

### CODE REVIEW

File: VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs [hunk 1]
```diff
@@ -2,72 +2,465 @@
...
+					return null;
+				return editor.GetDragPoint(controlId);
+			}
+
+			//Drag Points
+			[MenuItem(CONTROLPOINTS_MENUPATH + "/IsLocked", false, 1)]
+			private static void Lock(MenuCommand command)
+			{
+				ItemInspector editor = command.context as ItemInspector;
+				if (editor == null || editor.DragPointsEditor == null)
```
Comment (freezy) [inline VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs:78]: Out of interest, how can this happen?

File: VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs [hunk 2]
```diff
@@ -2,72 +2,465 @@
...
+							//Set Meshdirty to true there so it'll trigger again after Undo
+							List<Object> recordObjs = new List<Object>();
+							if (editable != null)
+							{
+								editable.MeshDirty = true;
+								recordObjs.Add(_itemInspector);
+							}
+							recordObjs.Add(_target as Behaviour);
+							_needMeshRebuilt = false;
+							Undo.RecordObjects(recordObjs.ToArray(), _pendingUndoMessage);
```
Comment (rbxnk) [inline VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs:422]: Undo isn't working for me, I wonder if there's something about delaying the record until repaint that unity doesn't like...
In either case it feels a little weird to rely on repaint for this, can we just record the change and trigger mesh rebuild in the end change check blocks?

File: VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs (hunk 1 above)
Comment (Vroonsh) [reply VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs:78]: Shouldn't...but you'll never know with asynchronous popmenu calls. Well...maybe overreacting :D

**Comment by Vroonsh** [comment]
@freezy you're definitely right about brackets on if statements, sometimes forgetting ;)

File: VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs (hunk 2 above)
Comment (Vroonsh) [reply VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs:422]: i'll put it back in the prepare, maybe the fix on the MeshExtensions side is enought for the mesh rebuild error.

File: VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs [hunk 3]
```diff
@@ -2,72 +2,465 @@
...
+
+			if (dpeditable.DragPointEditEnabled)
+			{
+				if (_foldoutControlPoints = EditorGUILayout.BeginFoldoutHeaderGroup(_foldoutControlPoints, "Drag Points"))
+				{
+					foreach (var cpoint in _controlPoints)
+					{
+						EditorGUILayout.LabelField("ID [" + cpoint.ControlId.ToString() + "]");
+						EditorGUILayout.Space(10);
+						EditorGUILayout.Toggle(cpoint.DragPoint.IsLocked, "IsLocked");
```
[the remark this answers is not in the document]
Comment (Vroonsh) [reply VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs:221]: Yeah it was in progress when i made the PR, i'll comment that for now

File: VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs (hunk 2 above)
Comment (Vroonsh) [reply VisualPinball.Unity/VisualPinball.Unity.Editor/Editors/DragPointsEditor.cs:422]: It fixes the Undo for Dragpoints but still no mesh rebuild after the Undo while i am putting both the ItemInspector and the ItemBehavior in the record.

**Comment by Vroonsh** [comment]
That's weird that my latest change on unity side failed a test on engine side

**Comment by freezy** [comment]
Flaky test, relaunched and looks good now.

**Comment by Vroonsh** [comment]
Seems the string interpolation syntax doesn't work in menu items brackets. Or I missed something.

**Comment by freezy** [comment]
@rbxnk is this fine now?

**Comment by freezy** [comment]
@Vroonsh could you address the height issue already?

**Comment by rbxnk** [comment]
@freezy Yeah looks good enough to me now, other than the height thing. @Vroonsh for that, we could just take the pre-change-check value for y (or z, I can't remember), and improve the handles in a future change

**Comment by freezy** [comment]
@Vroonsh Do you prefer fixing the height in this PR or in another one?

**Comment by Vroonsh** [comment]
Sorry guys it's nearly midnight there 😁, wasn't on the PC.
No I'll check that with the next one, already did some things on the inspector, I'll dig that during the week.

**Comment by freezy** [comment]
No worries :)

Good to merge then?

**Pull request merged**